### DISCUSSION TITLE
Internal Repository Not Reachable From Actions

### DISCUSSION THREAD

**Discussion by elihuntington** [community/community#26207] [Actions]
<p>I have a trio of internal repositories in my organization.</p>
<p>In the package.json of Repository A is a dependency on the contents of Repository B.  Repository C is just like B.</p>
<p>When I run yarn or npm locally, I have no problems with the installation of the dependencies.</p>
<p>However when I trigger a workflow that is intended to install dependencies then do a build, the process fails.</p>
<p>The output looks like this:</p>
<pre><code>yarn install v1.22.0
[1/4] Resolving packages...
error Command failed.
Exit code: 128
Command: git
Arguments: ls-remote --tags --heads ***github.com/&lt;myorg&gt;/&lt;RepositoryB&gt;.git
Directory: /github/workspace
Output:
remote: Repository not found.
fatal: repository 'https://github.com/&lt;myorg&gt;/&lt;RepositoryB&gt;.git/' not found
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
</code></pre>
<p>I have tried several approaches to fixing this including using a PAT with Secrets on Repository A. That PAT was given all permissions except for creating GPG keys, and Org Admin.  I have also tried using SSH Key and putting the private key in Repository A Secrets and the deploy key in Repository B.</p>
<p>This seems like a big missing feature of Actions if there is no way to accomplish.</p>

**Comment by logankilpatrick** [comment]
<p>Hey! Can you show the command that you are calling that gives the error? GitHub, in general, supports the ability to use a private repo as a dependency of another private repo so I think the culprit is the command itself. </p>

**Comment by elihuntington** [comment]
<pre><code>- name: Install Dependencies
              uses: &lt;myorg&gt;/&lt;myrepo&gt;/.github/actions-yarn@master
              with:
                  cmd: install
</code></pre>
<p>This is the command that fails.  I have tried the setup-node action as well.<br>
Inside my package.json file there are dependencies using OAUTH:</p>
<pre><code>"eslint-config": "git+https://&lt;OAUTH_TOKEN&gt;:x-oauth-basic@github.com/&lt;myorg&gt;/&lt;myrepo&gt;.git"
</code></pre>
<p>OAUTH_TOKEN is not a replace key in our system yet, so it’s right in there.  This works on development machines.<br>
Our org uses Single Sign On (SSO) as well if that’s a factor.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/elihuntington">@elihuntington</a> ,</p>
<p>I can reproduce the same with below code:</p>
<pre><code>steps:
      - name: checkout
        uses: actions/checkout@v2

      - name: gitremote
        run: |
          git ls-remote --tags --heads https://github.com/weide-zhou/ticket1.git
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181096058-a782cab4-bc42-46bb-84d0-0600b48384d4.png" alt="giterror.png" data-base62-sha1="1WVVuYL1vn1qALKKwD0RuOh4ceJ" width="575" height="185"></p>
<p>Did you use ‘actions/checkout@v2’ at prior step as well? ‘actions/checkout@v2’ will persist the GITHUB_TOKEN credential to the repo config to make git operations. Please use ‘actions/checkout@v1’ instead. Or if you use v2, please add ‘persist-credentials: false’ in the action args. Please check the <a href="https://github.community/t/action-does-not-trigger-another-on-push-tag-action/17148/4">link</a> for more details.</p>
<p>If the issue still persists, please share a repo sample or total yaml to help further investigation.</p>
<p>Thanks.</p>

**Comment by elihuntington** [comment]
<p>This did fix it for me!  Since I am in a private org, I had to clone the checkout/v1 action into my <code>.github</code> folder.<br>
I put Personal Access Token in GitHub Secrets and everything started working!  Thanks for your help.</p>

**Comment by SLG6658029** [comment]
<pre><code>yarn install v1.22.0
[1/4] Resolving packages...
error Command failed.
Exit code: 128
Command: git
Arguments: ls-remote --tags --heads ***github.com/&lt;myorg&gt;/&lt;RepositoryB&gt;.git
Directory: /github/workspace
Output:
remote: Repository not found.
fatal: repository 'https://github.com/&lt;myorg&gt;/&lt;RepositoryB&gt;.git/' not found
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
</code></pre>
